autogen/main/cluster.tf.tmpl

```diff
@@ -62,8 +62,14 @@ resource "google_container_cluster" "primary" {
 {% endif %}
   min_master_version = var.release_channel != null ? null : local.master_version
 
+{% if beta_cluster %}
+  cluster_telemetry {
+    type = var.cluster_telemetry_type
+  }
+{% else %}
```

Comment:
We may want to keep `logging_service` and `monitoring_service` to keep supporting 1.14 clusters as the API also supports these concurrently.

Reply:
Currently `cluster_telemetry` and `logging_service`/`monitoring_service` can't be set at the same time, see [this](https://github.com/GoogleCloudPlatform/magic-modules/blob/8a1b36217daf3d48ac49aba3769fb0872d0c0097/third_party/terraform/resources/resource_container_cluster.go.erb#L517) and [this](https://github.com/GoogleCloudPlatform/magic-modules/blob/8a1b36217daf3d48ac49aba3769fb0872d0c0097/third_party/terraform/resources/resource_container_cluster.go.erb#L730). Also from the output of the failed CI above, when I set both `cluster_telemetry` and `logging_service`/`monitoring_service`:
```
Error: ConflictsWith
       
         on ../../../modules/beta-public-cluster/cluster.tf line 62, in resource "google_container_cluster" "primary":
         62:   logging_service    = var.logging_service
       
       "logging_service": conflicts with cluster_telemetry
       
       
       Error: ConflictsWith
       
         on ../../../modules/beta-public-cluster/cluster.tf line 63, in resource "google_container_cluster" "primary":
         63:   monitoring_service = var.monitoring_service
       
       "monitoring_service": conflicts with cluster_telemetry
```

Reply:
Right, but the conditional should probably be based in the Terraform not the templating layer.

Reply:
Can you explain more? How can I use conditional in resource block? Is something like this ok?
```
{% if beta_cluster %}
  dynamic "cluster_telemetry" {
    for_each = var.cluster_telemetry_type != "" ? [1] : [0]
    content {
      type = var.cluster_telemetry_type
    }
  }
{% else %}
  logging_service    = var.logging_service
  monitoring_service = var.monitoring_service
{% endif %}
```

Reply:
I was thinking something more like this:

```
locals {
  cluster_telemetry_type_is_set = var.cluster_telemetry_type != ""
}

dynamic "cluster_telemetry" {
    for_each = local.cluster_telemetry_type_is_set > 0 ? [] : [1]
    content {
      type = var.cluster_telemetry_type
    }
  }
  logging_service    = local.cluster_telemetry_type_is_set ? null : var.logging_service
  monitoring_service =  local.cluster_telemetry_type_is_set ? null : var.monitoring_service
}
```

Reply:
I fixed as you suggested, but the CI is failed. Could you check what is the CI error?

Reply:
It looks like an unrelated ephemeral error, re-running.

Reply:
There might be an edge case where for 1.14 clusters you can use legacy `logging_service="logging.googleapis.com"` and this change will try to change to `cluster_telemetry="ENABLED"` since we set `cluster_telemetry_type` to `ENABLED` by default. However, this should be okay as it will be caught by the API and an error will be thrown.

Reply:
Hmm looks like we're getting an error on the sandbox-enabled-local example:

```
       
       Error: Required attribute is not set
       
         on ../../../modules/beta-public-cluster/cluster.tf line 22, in resource "google_container_cluster" "primary":
         22: resource "google_container_cluster" "primary" {
       
       
       
       Error: ConflictsWith
       
         on ../../../modules/beta-public-cluster/cluster.tf line 65, in resource "google_container_cluster" "primary":
         65:   logging_service    = local.cluster_telemetry_type_is_set ? null : var.logging_service
       
       "logging_service": conflicts with cluster_telemetry
       
       
       Error: ConflictsWith
       
         on ../../../modules/beta-public-cluster/cluster.tf line 66, in resource "google_container_cluster" "primary":
         66:   monitoring_service = local.cluster_telemetry_type_is_set ? null : var.monitoring_service
       
       "monitoring_service": conflicts with cluster_telemetry
```

Reply:
Hmm you have any idea to fix this?

Reply:
@nguyenhoaibao I think this should be 
```diff
- for_each = local.cluster_telemetry_type_is_set ? [1] : [0]
+ for_each = local.cluster_telemetry_type_is_set ? [1] : []
```
[0] is still a list with 1 element which is causing the creation of the dynamic block.